### PR TITLE
fix: Delete associated program message while pruning enrollments, events and teis

### PR DESCRIPTION
**Pull request by zubaira**
JIRA # [DHIS2-10810]

### CODE REVIEW

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java [hunk 1]
```diff
@@ -51,6 +51,11 @@
 public class JdbcMaintenanceStore
     implements MaintenanceStore
 {
+    private static final String DELETE_DELIVERY_CHANNELS = "delete from programmessage_deliverychannels where programmessagedeliverychannelsid in ";
+
+    private static final String DELETE_EMAIL = "delete from programmessage_emailaddresses where programmessageemailaddressid in ";
+
+    private static final String DELETE_PHONE_NUMBER = "delete from programmessage_phonenumbers where programmessagephonenumberid in ";
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java:58]: Although it is good to have constants, and this class could use a complete refactor, I think it is better to prefer consistency and use inline variables instead to make it similar to the existing code.

Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java]: resolved

File: dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java [hunk 2]
```diff
@@ -130,6 +137,7 @@ public int deleteSoftDeletedProgramInstances()
             "delete from programmessage_emailaddresses where programmessageemailaddressid in " + pmSelect,
             "delete from programmessage_phonenumbers where programmessagephonenumberid in " + pmSelect,
             "delete from programmessage where programinstanceid in " + piSelect,
+            "delete from programmessage where programstageinstanceid in " + psiSelect,
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java:140]: should it be
```suggestion
            "delete from programmessage where programinstanceid in " + piSelect,
```
??
or maybe it's not clear for me

**Review by gnespolino — APPROVED** [review]
Approved with just one some doubt

File: dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java (hunk 2 above)
Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java:140]: while deleting enrollment we also have to delete its associated event's program messages

**Pull request merged**